kms-message/src/kms_kmip_response_parser.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kms_message/kms_kmip_response_parser.h"
+
+#include "kms_endian_private.h"
+#include "kms_kmip_response_parser_private.h"
+#include "kms_message_private.h"
+#include "kms_request_str.h"
+
+struct _kms_kmip_response_parser_t {
+   uint32_t first_len;
+   uint32_t bytes_fed;
+   kms_request_str_t *buf;
+   bool failed;
+   char error[512];
+};
+
+/* FIRST_LENGTH_OFFSET is the offset of the first four byte length. */
+static const uint32_t FIRST_LENGTH_OFFSET = 4;
```

Comment:
```suggestion
/* FIRST_LENGTH_OFFSET is the offset of the first four-byte length field in a
 * message. */
static const uint32_t FIRST_LENGTH_OFFSET = 4;
```
Not clear what "first four byte length" is referring to. I assume it means the first length field in a message, which is 4 bytes large?